topdown/eval.go

```diff
@@ -2861,6 +2898,82 @@ func (e evalVirtualPartial) evalCache(iter unifyIterator) (evalVirtualPartialCac
 	return hint, nil
 }
 
+// vcKeyScope represents the scoping that pre-rule-eval ref unification imposes on a virtual cache entry.
+type vcKeyScope struct {
+	ast.Ref
```

Comment:
Would we safe memory if we kept the index next to the ast.Ref?

```go
type vcKeyScope struct {
  ref ast.Ref
  idx int
}
```
so that we can avoid copying the `ast.Ref`? I guess it doesn't make a huge difference 🤔